app/src/main/java/com/duckduckgo/app/browser/BrowserActivity.kt

```diff
@@ -52,7 +52,7 @@ import org.jetbrains.anko.longToast
 import timber.log.Timber
 import javax.inject.Inject
 
-class BrowserActivity : DuckDuckGoActivity(), CoroutineScope by MainScope() {
+class BrowserActivity : DuckDuckGoActivity(), DaxDialogListener, CoroutineScope by MainScope() {
```

Comment:
I've moved the listener here so it's easier to know if the dax dialog is shown or not. Keeping this in the fragment means having to check every single fragment to know if the dialog is being shown to remove it. By doing that there is also the risk of an IllegalStateException (since we do this onCreate) because some fragments might not be attached yet at that point.

Reply:
I'm worried about that `currentTab` reference since we are not accessing by tab, and we are keeping the reference, I don't know if we should rely so strong on it for this to work.
The problem that you are describing is because multiple fragments trying to remove the same dialog?